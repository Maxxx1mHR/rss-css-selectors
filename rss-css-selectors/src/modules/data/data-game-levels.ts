// export const dataExampleLayout = {
//   1: `<round data-id="1" class="shape round" data-tooltip="<round></round>"></round>
//       <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>`,

//   2: `<square data-id="1" class="shape square"  data-tooltip="<square></square>"></square>
//       <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>
//       <square data-id="3" class="shape square" data-tooltip="<square></square>"></square>`,

//   3: `<round data-id="1" id="border" class="shape round border" data-tooltip='<round id="border"></round>'></round>
//       <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>
//       <square data-id="3" class="shape square" data-tooltip="<square></square>"></square>`,

//   4: `<square data-id="1" class="shape square" data-tooltip="<square></square>"></square>
//       <round data-id="2" class="shape round" data-tooltip="<round></round>">
//         <point data-id="3" class="shape point" data-tooltip="<point></point>"></point>
//       </round>
//       <point data-id="4" class="shape point" data-tooltip="<point></point>"></point>`,

//   5: `<square data-id="1" class="shape square" data-tooltip="<square></square>">
//         <blue data-id="2" class="shape blue" data-tooltip="<blue></blue>"></blue>
//       </square>
//       <round data-id="3" id="border" class="shape round border" data-tooltip='<round id="border"></round>'>
//         <rectangle data-id="4" class="shape rectangle" data-tooltip="<rectangle></rectangle>"></rectangle>
//       </round>
//       <round data-id="5" class="shape round" data-tooltip="<round></round>">
//         <rectangle data-id="6" class="shape rectangle" data-tooltip="<rectangle></rectangle>"></rectangle >
//       </round>
//       `,
// };

// export const dataHTMLViewer = {
//   1: `
//   &ltdiv class="boat"&gt
//     <round data-id="1" class="code">&ltround/&gt</round>
//     <round data-id="2" class="code">&ltround/&gt</round>
//   &lt/div&gt
//   `,
//   2: `
//   &ltdiv class="boat"&gt
//     <square data-id="1" class="code">&ltsquare/&gt</square>
//     <round data-id="2" class="code">&ltround/&gt</round>
//     <square data-id="3" class="code">&ltsquare/&gt</square>
//   &lt/div&gt
//   `,
//   3: `
//   &ltdiv class="boat"&gt
//     <square data-id="1" id="border" class="code">&ltsquare id="border"/&gt</square>
//     <round data-id="2" class="code">&ltround/&gt</round>
//     <square data-id="3" class="code">&ltsquare/&gt</square>
//   &lt/div&gt
//   `,
//   4: `
//   &ltdiv class="boat"&gt
//     <square data-id="1" class="code">&ltsquare/&gt</square>
//     <round data-id="2" class="code">&ltround&gt
//       <point data-id="3" class="code">&ltpoint/&gt</point>
//     &lt/round&gt</round>
//     <point data-id="4" class="code">&ltpoint/&gt</point>
//   &lt/div&gt`,
//   5: `
//   &ltdiv class="boat"&gt
//     <square data-id="1" class="code">&ltsquare&gt
//       <blue data-id="2" class="code">&ltblue/&gt</blue>
//     &ltsquare/&gt</square><round data-id="3" id="border" class="code">
//     &ltround id="border"&gt
//       <rectangle data-id="4" class="code">&ltrectangle/&gt</rectangle>
//     &lt/round&gt</round>
//     <round data-id="5" class="code">&ltround&gt
//       <rectangle data-id="6" class="code">&lt/rectangle/&gt</rectangle >
//     &lt/round&gt</round>
//   &lt/div&gt`,
// };

export const dataGameLevels = [
  {
    level: 1,
    taskName: 'Type Selector',
    taskDescription: 'Select elements by their type',
    attribute: `<span>A</span>`,
    correction:
      'Selects all elements of type <span>A</span>. Type refers to the type of tag, so <span>&ltdiv&gt</span>, <span>&ltp&gt</span> and <span>&ltul&gt</span> are all different element types.',
    example: '<span>div</span> selects all <span>&ltdiv&gt</span> elements',
    dataExampleLayout: `<wheel data-id="1" class="shape wheel" data-tooltip="<wheel></wheel>"></wheel>
      <wheel data-id="2" class="shape wheel" data-tooltip="<wheel></wheel>"></wheel>`,
    dataHTMLViewer: `
    &ltdiv class="boat"&gt
    <wheel data-id="1" class="code">&wheel/&gt</wheel>
    <wheel data-id="2" class="code">&wheel/&gt</wheel>
  &lt/div&gt
      `,
  },
  {
    level: 2,
    taskName: 'Type Selector',
    taskDescription: 'Select elements by their type',
    attribute: `<span>A</span>`,
    correction:
      'Selects all elements of type <span>A</span>. Type refers to the type of tag, so <span>&ltdiv&gt</span>, <span>&ltp&gt</span> and <span>&ltul&gt</span> are all different element types.',
    example: '<span>div</span> selects all <span>&ltdiv&gt</span> elements',
    dataExampleLayout: `<map data-id="1" class="shape map"  data-tooltip="<map></map>"></map>
    <wheel data-id="2" class="shape wheel" data-tooltip="<wheel></wheel>"></wheel>
    <map data-id="3" class="shape map" data-tooltip="<map></map>"></map>`,
    dataHTMLViewer: `
    &ltdiv class="boat"&gt
    <map data-id="1" class="code">&map/&gt</map>
    <wheel data-id="2" class="code">&wheel/&gt</wheel>
    <map data-id="3" class="code">&map/&gt</map>
  &lt/div&gt
      `,
  },
  {
    level: 3,
    taskName: 'ID Selector',
    taskDescription: 'Select elements with an ID  ',
    attribute: `<span>#id</span>`,
    correction:
      'Selects the element with a specific <span>id</span>. You can also combine the ID selector with the type selector.',
    example: '<span>#cool</span> selects any element with <span>id="cool"</span> elements',
    dataExampleLayout: `<wheel data-id="1" id="first" class="shape wheel first" data-tooltip='<wheel id="first"></wheel>'></wheel>
      <wheel data-id="2" class="shape wheel" data-tooltip="<wheel></wheel>"></wheel>
      <map data-id="3" class="shape map" data-tooltip="<map></map>"></map>`,
    dataHTMLViewer: `
    &ltdiv class="boat"&gt
    <wheel data-id="1" id="first" class="code">&wheel id="first"/&gt</wheel>
    <wheel data-id="2" class="code">&wheel/&gt</wheel>
    <map data-id="3" class="code">&map/&gt</map>
  &lt/div&gt
      `,
  },
  {
    level: 4,
    taskName: 'Descendant Selector',
    taskDescription: 'Select an element inside another element',
    attribute: `<span>A  B</span>`,
    correction:
      'Selects all <span>B</span> inside of <span>A</span>. <span>B</span> is called a descendant because it is inside of another element.',
    example:
      '<span>p strong </span> selects all <span>&ltstrong&gt</span> elements that are inside of any <span>&lp&gt</span>',
    dataExampleLayout: `<map data-id="1" class="shape map" data-tooltip="<map></map>"></map>
    <wheel data-id="2" class="shape wheel" data-tooltip="<wheel></wheel>">
      <spyglass data-id="3" class="shape spyglass" data-tooltip="<spyglass></spyglass>"></spyglass>
    </wheel>
    <spyglass data-id="4" class="shape spyglass" data-tooltip="<spyglass></spyglass>"></spyglass>`,
    dataHTMLViewer: `
    &ltdiv class="boat"&gt
    <map data-id="1" class="code">&map/&gt</map>
    <wheel data-id="2" class="code">&wheel&gt
      <spyglass data-id="3" class="code">&spyglass/&gt</spyglass>
    &lt/wheel&gt</wheel>
    <spyglass data-id="4" class="code">&spyglass/&gt</spyglass>
  &lt/div&gt
      `,
  },
  {
    level: 5,
    taskName: 'No Name...',
    taskDescription: 'Combine the Descendant & ID Selectors',
    attribute: `<span>#id  A</span>`,
    correction: 'You can combine any selector with the descendent selector.',
    example:
      '<span>#cool span</span> selects all <span>&ltspan&gt</span> elements  that are inside of elements with <span>id="cool"</span>',
    dataExampleLayout: `<map data-id="1" class="shape map" data-tooltip="<map></map>">
    <sword data-id="2" class="shape sword" data-tooltip="<sword></sword>"></sword>
  </map>
  <wheel data-id="3" id="border" class="shape wheel border" data-tooltip='<wheel id="border"></wheel>'>
    <bottle data-id="4" class="shape bottle" data-tooltip="<bottle></bottle>"></bottle>
  </wheel>
  <wheel data-id="5" class="shape wheel" data-tooltip="<wheel></wheel>">
    <bottle data-id="6" class="shape bottle" data-tooltip="<bottle></bottle>"></bottle >
  </wheel>`,
    dataHTMLViewer: `
    &ltdiv class="boat"&gt
    <map data-id="1" class="code">&map&gt
      <sword data-id="2" class="code">&sword/&gt</sword>
    &map/&gt</map><wheel data-id="3" id="border" class="code">
    &wheel id="border"&gt
      <bottle data-id="4" class="code">&bottle/&gt</bottle>
    &lt/wheel&gt</wheel>
    <wheel data-id="5" class="code">&wheel&gt
      <bottle data-id="6" class="code">&lt/bottle/&gt</bottle >
    &lt/wheel&gt</wheel>
  &lt/div&gt
      `,
  },
];
