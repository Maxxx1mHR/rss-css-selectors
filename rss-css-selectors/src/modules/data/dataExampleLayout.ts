export const dataExampleLayout = {
  1: `<round data-id="1" class="shape round" data-tooltip="<round></round>"></round>
      <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>`,

  2: `<square data-id="1" class="shape square"  data-tooltip="<square></square>"></square>
      <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>
      <square data-id="3" class="shape square" data-tooltip="<square></square>"></square>`,

  3: `<round data-id="1" id="border" class="shape round border" data-tooltip='<round id="border"></round>'></round>
      <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>
      <square data-id="3" class="shape square" data-tooltip="<square></square>"></square>`,

  4: `<square data-id="1" class="shape square" data-tooltip="<square></square>"></square>
      <round data-id="2" class="shape round" data-tooltip="<round></round>">
        <point data-id="3" class="shape point" data-tooltip="<point></point>"></point>
      </round>
      <point data-id="4" class="shape point" data-tooltip="<point></point>"></point>`,

  5: `<square data-id="1" class="shape square" data-tooltip="<square></square>">
        <blue data-id="2" class="shape blue" data-tooltip="<blue></blue>"></blue>
      </square>
      <round data-id="3" id="border" class="shape round border" data-tooltip='<round id="border"></round>'>
        <rectangle data-id="4" class="shape rectangle" data-tooltip="<rectangle></rectangle>"></rectangle>
      </round>
      <round data-id="5" class="shape round" data-tooltip="<round></round>">
        <rectangle data-id="6" class="shape rectangle" data-tooltip="<rectangle></rectangle>"></rectangle >
      </round>
      `,
};

export const dataHTMLViewer = {
  1: `
  &ltdiv class="boat"&gt
    <round data-id="1" class="code">&ltround/&gt</round>
    <round data-id="2" class="code">&ltround/&gt</round>
  &lt/div&gt
  `,
  2: `
  &ltdiv class="boat"&gt
    <square data-id="1" class="code">&ltsquare/&gt</square>
    <round data-id="2" class="code">&ltround/&gt</round>
    <square data-id="3" class="code">&ltsquare/&gt</square>
  &lt/div&gt
  `,
  3: `
  &ltdiv class="boat"&gt
    <square data-id="1" id="border" class="code">&ltsquare id="border"/&gt</square>
    <round data-id="2" class="code">&ltround/&gt</round>
    <square data-id="3" class="code">&ltsquare/&gt</square>
  &lt/div&gt
  `,
  4: `
  &ltdiv class="boat"&gt
    <square data-id="1" class="code">&ltsquare/&gt</square>
    <round data-id="2" class="code">&ltround&gt
      <point data-id="3" class="code">&ltpoint/&gt</point>
    &lt/round&gt</round>
    <point data-id="4" class="code">&ltpoint/&gt</point>
  &lt/div&gt`,
  5: `
  &ltdiv class="boat"&gt
    <square data-id="1" class="code">&ltsquare&gt
      <blue data-id="2" class="code">&ltblue/&gt</blue>
    &ltsquare/&gt</square><round data-id="3" id="border" class="code">
    &ltround id="border"&gt
      <rectangle data-id="4" class="code">&ltrectangle/&gt</rectangle>
    &lt/round&gt</round>
    <round data-id="5" class="code">&ltround&gt
      <rectangle data-id="6" class="code">&lt/rectangle/&gt</rectangle >
    &lt/round&gt</round>
  &lt/div&gt`,
};

export const dataLevelsGame = [
  {
    level: 1,
    taskName: 'Type Selector',
    taskDescription: 'Select elements by their type',
    attribute: 'A',
    correction:
      'Selects all elements of type A. Type refers to the type of tag, so div, p and ul are all different element types.',
    example: 'div selects all div elements.',
    dataExampleLayout: `<round data-id="1" class="shape round" data-tooltip="<round></round>"></round>
      <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>`,
    dataHTMLViewer: `
      &ltdiv class="boat"&gt
        <round data-id="1" class="code">&ltround/&gt</round>
        <round data-id="2" class="code">&ltround/&gt</round>
      &lt/div&gt
      `,
  },
  {
    level: 2,
    taskName: 'Type Selector',
    taskDescription: 'Select elements by their type',
    attribute: 'A',
    correction:
      'Selects all elements of type A. Type refers to the type of tag, so div, p and ul are all different element types.',
    example: 'div selects all div elements.',
    dataExampleLayout: `<square data-id="1" class="shape square"  data-tooltip="<square></square>"></square>
    <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>
    <square data-id="3" class="shape square" data-tooltip="<square></square>"></square>`,
    dataHTMLViewer: `
    &ltdiv class="boat"&gt
      <square data-id="1" id="border" class="code">&ltsquare id="border"/&gt</square>
      <round data-id="2" class="code">&ltround/&gt</round>
      <square data-id="3" class="code">&ltsquare/&gt</square>
    &lt/div&gt
    `,
  },
  {
    level: 3,
    taskName: 'ID Selector',
    taskDescription: 'Select elements with an ID',
    attribute: '#id',
    correction: 'Selects the element with a specific id. You can also combine the ID selector with the type selector.',
    example: '#cool selects any element with id="cool"',
    dataExampleLayout: `<round data-id="1" id="border" class="shape round border" data-tooltip='<round id="border"></round>'></round>
    <round data-id="2" class="shape round" data-tooltip="<round></round>"></round>
    <square data-id="3" class="shape square" data-tooltip="<square></square>"></square>`,
    dataHTMLViewer: `
    &ltdiv class="boat"&gt
      <square data-id="1" id="border" class="code">&ltsquare id="border"/&gt</square>
      <round data-id="2" class="code">&ltround/&gt</round>
      <square data-id="3" class="code">&ltsquare/&gt</square>
    &lt/div&gt
    `,
  },
];
