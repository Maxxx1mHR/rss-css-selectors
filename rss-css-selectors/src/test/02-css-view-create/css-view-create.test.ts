// import { dataGameLevels } from '../../modules/data/data-game-levels';
// import CssView from '../../modules/view/css-view';

// describe('render block css', () => {
//   const level = 0;
//   it('renders correctly', () => {
//     const cssView = new CssView(dataGameLevels, level);
//     const css = cssView.createView();
//     expect(css).toMatchSnapshot();
//   });
// });

import { dataGameLevels } from '../../modules/data/data-game-levels';
import HtmlView from '../../modules/view/html-view';

describe('render block boat for each level', () => {
  const countLevel = dataGameLevels.length;
  for (let i = 0; i < countLevel; i++) {
    it(`renders level ${i} correctly`, () => {
      const htmlView = new HtmlView(dataGameLevels, i);
      const html = htmlView.createView(dataGameLevels, i);
      expect(html).toMatchSnapshot();
    });
  }
});
