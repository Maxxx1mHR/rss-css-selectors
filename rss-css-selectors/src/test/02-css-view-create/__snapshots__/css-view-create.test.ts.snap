// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render block boat for each level renders level 0 correctly 1`] = `
<code
  class="editor__code-html"
>
  
    &lt;div class="boat"&gt;
        
  <wheel
    class="code"
    data-id="1"
  >
    &lt;wheel/&gt;
  </wheel>
  
        
  <wheel
    class="code"
    data-id="2"
  >
    &lt;wheel/&gt;
  </wheel>
  
    &lt;/div&gt;
      
</code>
`;

exports[`render block boat for each level renders level 1 correctly 1`] = `
<code
  class="editor__code-html"
>
  
    &lt;div class="boat"&gt;
        
  <map
    class="code"
    data-id="1"
  >
    &lt;map/&gt;
  </map>
  
        
  <wheel
    class="code"
    data-id="2"
  >
    &lt;wheel/&gt;
  </wheel>
  
        
  <map
    class="code"
    data-id="3"
  >
    &lt;map/&gt;
  </map>
  
    &lt;/div&gt;
      
</code>
`;

exports[`render block boat for each level renders level 2 correctly 1`] = `
<code
  class="editor__code-html"
>
  
    &lt;div class="boat"&gt;
        
  <wheel
    class="code"
    data-id="1"
    id="first"
  >
    &lt;wheel id="first"/&gt;
  </wheel>
  
        
  <wheel
    class="code"
    data-id="2"
  >
    &lt;wheel/&gt;
  </wheel>
  
        
  <map
    class="code"
    data-id="3"
  >
    &lt;map/&gt;
  </map>
  
    &lt;/div&gt;
      
</code>
`;

exports[`render block boat for each level renders level 3 correctly 1`] = `
<code
  class="editor__code-html"
>
  
    &lt;div class="boat"&gt;
        
  <map
    class="code"
    data-id="1"
  >
    &lt;map/&gt;
  </map>
  
        
  <wheel
    class="code"
    data-id="2"
  >
    &lt;wheel&gt;
            
    <spyglass
      class="code"
      data-id="3"
    >
      &lt;spyglass/&gt;
    </spyglass>
    
        &lt;/wheel&gt;
  </wheel>
  
        
  <spyglass
    class="code"
    data-id="4"
  >
    &lt;spyglass/&gt;
  </spyglass>
  
    &lt;/div&gt;
      
</code>
`;

exports[`render block boat for each level renders level 4 correctly 1`] = `
<code
  class="editor__code-html"
>
  
    &lt;div class="boat"&gt;
        
  <map
    class="code"
    data-id="1"
  >
    &lt;map&gt;
            
    <sword
      class="code"
      data-id="2"
    >
      &lt;sword/&gt;
    </sword>
    
        &lt;/map&gt;
  </map>
  <wheel
    class="code"
    data-id="3"
    id="captain"
  >
    
        &lt;wheel id="captain"&gt;
            
    <bottle
      class="code"
      data-id="4"
    >
      &lt;bottle/&gt;
    </bottle>
    
        &lt;/wheel&gt;
  </wheel>
  
        
  <wheel
    class="code"
    data-id="5"
  >
    &lt;wheel&gt;
            
    <bottle
      class="code"
      data-id="6"
    >
      &lt;bottle/&gt;
    </bottle>
    
        &lt;/wheel&gt;
  </wheel>
  
    &lt;/div&gt;
      
</code>
`;

exports[`render block boat for each level renders level 5 correctly 1`] = `
<code
  class="editor__code-html"
>
  
    &lt;div class="boat"&gt;
        
  <wheel
    class="code"
    data-id="1"
  >
    &lt;wheel&gt;
            
    <spyglass
      class="code"
      data-attr="spyglass"
      data-id="2"
    >
      &lt;spyglass data-attr="spyglass"/&gt;
    </spyglass>
    
            
    <spyglass
      class="code"
      data-id="3"
    >
      &lt;spyglass /&gt;
    </spyglass>
    
        &lt;/wheel&gt;
  </wheel>
  
        
  <map
    class="code"
    data-id="4"
  >
    &lt;map&gt;
            
    <sword
      class="code"
      data-attr="first-sword"
      data-id="5"
    >
      &lt;sword data-attr="first-sword"/&gt;
    </sword>
    
            
    <sword
      class="code"
      data-attr="second"
      data-id="6"
    >
      &lt;sword data-attr="second"/&gt;
    </sword>
    
            
    <sword
      class="code"
      data-attr="sword-third"
      data-id="7"
    >
      &lt;sword data-attr="sword-third"/&gt;
    </sword>
    
        &lt;/map&gt;
        
  </map>
  <wheel
    class="code"
    data-id="8"
  >
    &lt;wheel&gt;
            
    <bottle
      class="code"
      data-attr="frontend"
      data-id="9"
    >
      &lt;bottle data-attr="frontend"/&gt;
    </bottle>
    
            
    <bottle
      class="code"
      data-attr="bottle"
      data-id="10"
    >
      &lt;bottle data-attr="bottle"/&gt;
    </bottle>
    
            
    <bottle
      class="code"
      data-attr="backend"
      data-id="11"
    >
      &lt;bottle data-attr="backend"/&gt;
    </bottle>
    
        &lt;/wheel&gt;
        
  </wheel>
  <spyglass
    class="code"
    data-attr="spyglass"
    data-id="12"
  >
    &lt;spyglass data-attr="spyglass"/&gt;
  </spyglass>
  
    &lt;/div&gt;
    
</code>
`;

exports[`render block boat for each level renders level 6 correctly 1`] = `
<code
  class="editor__code-html"
>
  
    &lt;div class="boat"&gt;
        
  <map
    class="code"
    data-id="1"
  >
    &lt;map&gt;
            
    <bottle
      class="code"
      data-id="2"
    >
      &lt;bottle/&gt;
    </bottle>
    
        &lt;/map&gt;
  </map>
  
        
  <wheel
    class="code"
    data-id="3"
  >
    &lt;wheel&gt;
            
    <bottle
      class="code"
      data-id="4"
    >
      &lt;bottle/&gt;
    </bottle>
    
        &lt;/wheel&gt;
  </wheel>
  
        
  <wheel
    class="code"
    data-id="5"
  >
    &lt;wheel&gt;
            
    <bottle
      class="code"
      data-id="6"
    >
      &lt;bottle/&gt;
    </bottle>
    
            
    <bottle
      class="code"
      data-id="7"
    >
      &lt;bottle/&gt;
    </bottle>
    
        &lt;/wheel&gt;
  </wheel>
  
    &lt;/div&gt;
    
</code>
`;

exports[`render block boat for each level renders level 7 correctly 1`] = `
<code
  class="editor__code-html"
>
  
    &lt;div class="boat"&gt;
        
  <map
    class="code"
    data-id="1"
  >
    &lt;map&gt;
            
    <spyglass
      class="code"
      data-id="2"
    >
      &lt;spyglass/&gt;
    </spyglass>
    
            
    <spyglass
      class="code"
      data-id="3"
    >
      &lt;spyglass/&gt;
    </spyglass>
    
        &lt;/map&gt;
  </map>
  
        
  <map
    class="code"
    data-id="4"
  >
    &lt;map&gt;
            
    <spyglass
      class="code"
      data-id="5"
    >
      &lt;spyglass/&gt;
    </spyglass>
    
            
    <spyglass
      class="code"
      data-id="6"
    >
      &lt;spyglass/&gt;
    </spyglass>
    
        &lt;/map&gt;
  </map>
  
    &lt;/div&gt;
    
</code>
`;

exports[`render block boat for each level renders level 8 correctly 1`] = `
<code
  class="editor__code-html"
>
  
    &lt;div class="boat"&gt;
        
  <map
    class="code"
    data-id="1"
  >
    &lt;map&gt;
            
    <wheel
      class="code"
      data-id="2"
    >
      &lt;wheel&gt;
                
      <bottle
        class="code"
        data-attr="frontend"
        data-id="3"
      >
        &lt;bottle/&gt;
      </bottle>
      
                
      <bottle
        class="code"
        data-attr="bottle"
        data-id="4"
      >
        &lt;bottle/&gt;
      </bottle>
      
                
      <bottle
        class="code"
        data-attr="backend"
        data-id="5"
      >
        &lt;bottle/&gt;
      </bottle>
      
                
      <bottle
        class="code"
        data-attr="backend"
        data-id="6"
      >
        &lt;bottle/&gt;
      </bottle>
      
            &lt;/wheel&gt;
    </wheel>
    
        &lt;/map&gt;
  </map>
  
        
  <map
    class="code"
    data-id="7"
  >
    &lt;map&gt;
            
    <wheel
      class="code"
      data-id="8"
    >
      &lt;wheel id="#captain"&gt;
                
      <bottle
        class="code"
        data-attr="frontend"
        data-id="9"
      >
        &lt;bottle/&gt;
      </bottle>
      
                
      <bottle
        class="code"
        data-attr="bottle"
        data-id="10"
      >
        &lt;bottle/&gt;
      </bottle>
      
                
      <bottle
        class="code"
        data-attr="backend"
        data-id="11"
      >
        &lt;bottle/&gt;
      </bottle>
      
                
      <bottle
        class="code"
        data-attr="backend"
        data-id="12"
      >
        &lt;bottle/&gt;
      </bottle>
      
            &lt;/wheel&gt;
        
    </wheel>
    &lt;/map&gt;
  </map>
  
    &lt;/div&gt;
    
</code>
`;

exports[`render block boat for each level renders level 9 correctly 1`] = `
<code
  class="editor__code-html"
>
  
    &lt;div class="boat"&gt;
        
  <map
    class="code"
    data-id="1"
  >
    &lt;map&gt;
            
    <bottle
      class="code"
      data-id="2"
    >
      &lt;bottle/&gt;
    </bottle>
    
            
    <bottle
      class="code"
      data-id="3"
    >
      &lt;bottle/&gt;
    </bottle>
    
            
    <bottle
      class="code"
      data-id="4"
    >
      &lt;bottle/&gt;
    </bottle>
    
            
    <bottle
      class="code"
      data-id="5"
    >
      &lt;bottle/&gt;
    </bottle>
    
        &lt;/map&gt;
  </map>
  
        
  <wheel
    class="code"
    data-id="6"
  >
    &lt;wheel id="captain"&gt;
            
    <bottle
      class="code"
      data-id="7"
    >
      &lt;bottle/&gt;
    </bottle>
    
            
    <bottle
      class="code"
      data-id="8"
    >
      &lt;bottle/&gt;
    </bottle>
    
            
    <bottle
      class="code"
      data-id="9"
    >
      &lt;bottle/&gt;
    </bottle>
    
        &lt;/wheel&gt;
  </wheel>
  
        
  <wheel
    class="code"
    data-id="10"
  >
    &lt;wheel&gt;
            
    <spyglass
      class="code"
      data-id="11"
    >
      &lt;spyglass/&gt;
    </spyglass>
    
            
    <bottle
      class="code"
      data-id="12"
    >
      &lt;bottle/&gt;
    </bottle>
    
            
    <bottle
      class="code"
      data-id="13"
    >
      &lt;bottle/&gt;
    </bottle>
    
        &lt;/wheel&gt;
  </wheel>
  
    &lt;/div&gt;
    
</code>
`;
